core/client/fs/src/main/java/alluxio/client/file/cache/PageStore.java

```diff
@@ -69,7 +85,64 @@ static PageStore create(AlluxioConfiguration conf) {
         throw new IllegalArgumentException(String.format("Unrecognized store type %s",
             storeType.name()));
     }
-    options.setRootDir(conf.get(PropertyKey.USER_CLIENT_CACHE_DIR));
+    String rootPath = conf.get(PropertyKey.USER_CLIENT_CACHE_DIR);
+    options.setRootDir(rootPath);
+    Path confPath = Paths.get(rootPath, CONF_FILE);
+    boolean canLoad = false;
+    if (Files.exists(confPath)) {
+      Properties properties = ConfigurationUtils.loadPropertiesFromFile(confPath.toString());
+      if (properties != null) {
+        AlluxioProperties alluxioProperties = new AlluxioProperties();
+        alluxioProperties.merge(properties, Source.DEFAULT);
+        AlluxioConfiguration cacheConf = new InstancedConfiguration(alluxioProperties);
+        // check store type
+        if (cacheConf.get(PropertyKey.USER_CLIENT_CACHE_STORE_TYPE).equals(
+            conf.get(PropertyKey.USER_CLIENT_CACHE_STORE_TYPE))
+            // check page size
+            && cacheConf.getBytes(PropertyKey.USER_CLIENT_CACHE_PAGE_SIZE)
+            == conf.getBytes(PropertyKey.USER_CLIENT_CACHE_PAGE_SIZE)
+            // check enough cache size
+            && cacheConf.getBytes(PropertyKey.USER_CLIENT_CACHE_SIZE)
+            <= conf.getBytes(PropertyKey.USER_CLIENT_CACHE_SIZE)
+            // check alluxio version
+            && cacheConf.get(PropertyKey.VERSION).equals(
+            conf.get(PropertyKey.VERSION))) {
+          LOG.info("Found recoverable local cache at {}", rootPath);
+          canLoad = true;
+        } else {
+          LOG.info("Found local cache at {} with incompatible configuration.", rootPath);
+        }
+      }
+    }
+    if (!canLoad) {
+      LOG.info("Clean cache directory {}", rootPath);
+      File rootDir = new File(rootPath);
+      try {
+        if (Files.isDirectory(rootDir.toPath())) {
+          FileUtils.deleteDirectory(rootDir);
+        }
+        FileUtils.forceMkdir(rootDir);
+      } catch (IOException e) {
+        throw new IllegalStateException(
+            String.format("failed to clean cache directory %s", rootDir), e);
+      }
+      Properties properties = new Properties();
+      PropertyKey[] keys = new PropertyKey[]{
+          PropertyKey.USER_CLIENT_CACHE_STORE_TYPE,
+          PropertyKey.USER_CLIENT_CACHE_PAGE_SIZE,
+          PropertyKey.USER_CLIENT_CACHE_SIZE,
+          PropertyKey.VERSION
+      };
+      for (PropertyKey key : keys) {
+        properties.setProperty(key.getName(), conf.get(key));
+      }
+      try (FileOutputStream stream = new FileOutputStream(confPath.toString())) {
+        properties.store(stream, "Alluxio local cache configuration");
+      } catch (IOException e) {
+        throw new IllegalStateException(
+            String.format("failed to write cache configuration to file %s", confPath), e);
+      }
+    }
```

Comment:
It feels kind of weird to not do the loading here. Also, could you put the clean up/load compatibility logic in another method, this method is doing a lot.

Reply:
I agree it feels weird. However the load method needs to pass a list of page ids to the meta store which cannot be returned in this factory method.

Reply:
Extracted format and check to separate methods.